x-pack/plugins/security_solution/server/endpoint/services/actions/clients/sentinelone/sentinel_one_actions_client.ts

```diff
@@ -195,28 +200,54 @@ export class SentinelOneActionsClient extends ResponseActionsClientImpl {
       },
     });
 
-    return this.fetchActionDetails(actionRequestDoc.EndpointActions.action_id);
-  }
-
-  async release(options: IsolationRouteRequestBody): Promise<ActionDetails> {
-    await this.validateRequest(options);
-    await this.sendAction(SUB_ACTION.RELEASE_HOST, {
-      uuid: options.endpoint_ids[0],
-    });
-
-    const actionRequestDoc = await this.writeActionRequestToEndpointIndex({
+    const createPayload: CreateActionPayload = {
       ...options,
-      command: 'unisolate',
-    });
+      command: commandName,
+      action_id: actionRequestDoc.EndpointActions.action_id,
+      user: { username: this.actionsClientOptions.username },
+    };
 
-    await this.writeActionResponseToEndpointIndex({
-      actionId: actionRequestDoc.EndpointActions.action_id,
-      agentId: actionRequestDoc.agent.id,
-      data: {
-        command: actionRequestDoc.EndpointActions.data.command,
-      },
-    });
+    try {
+      const agentId = options.endpoint_ids[0];
+      await updateCases({
+        casesClient: this.options.casesClient,
+        endpointData: [
+          {
+            host: {
+              hostname: actionRequestDoc.EndpointActions.data.hosts?.[agentId].name as string,
+            },
+            agent: {
+              id: agentId,
+              type: options.agent_type || 'endpoint',
+            },
+          },
+        ],
+        createActionPayload: createPayload,
+      });
+    } catch (err) {
+      // failures during update of cases should not cause the response action to fail. Just log error
+      this.log.warn(`failed to update cases: ${err.message}\n${stringify(err)}`);
+    }
 
     return this.fetchActionDetails(actionRequestDoc.EndpointActions.action_id);
```

Comment:
This is going to cause Type problems once we introduce other response actions. `handleResponseAction()` will likely need to be a Generic that enables the called to define the narrow type for `ActionDetails` that is returned. (Ex. see the Endpoint response actions client)

Reply:
I removed the handleResponseAction function from S1, now just passing this.updateCases separately per command. 
I hope it looks better now ;)

Reply:
Thanks. That will help and we'll have an opportunity to consolidate redundant code once we get to implementation of a more complex response action.